limiter/limiter.go

```diff
@@ -0,0 +1,54 @@
+package limiter
+
+import "time"
+
+type LeakyBucket struct {
+	bucket chan struct{}
+	quit   chan struct{}
+}
+
+func (rl *LeakyBucket) fill(burst int) {
+	for i := 0; i < burst; i++ {
+		rl.bucket <- struct{}{}
+	}
+}
+
+func (rl *LeakyBucket) drip(drip float64) {
+	driptime := float64(1000000000) / drip
```

Comment:
Basically I take one second expressed in nanoseconds (1000000000).

Then divide that by the number of calls someone wishes to make in 1 second.

Then sleep for that duration in nanoseconds.

If it's not clear, then i should do more to make it read better. But i think the basic approach is ok?